adobe/react-spectrum
# Fix double onChangeEnd in ColorWheel

Closes #2247 

Seems like this may have been a stale value issue with useState via setDragging. Using a ref for the dragging state seems to solve this issue.

PR also includes a commit to check for pointer events and conditionally add event listeners (not directly related to this issue, so that can be reverted if needed).

## ✅ Pull Request Checklist:


## 📝 Test Instructions:

Check ColorWheel's 'controlled hsl' story:

- Verify that `onChangeEnd` is only called once on track click and thumb drag events. Previously, on a quick click on the track,`onChangeEnd` would be called twice intermittently.

- Verify that that there is no longer a console warning from onChangeEnd. Previously there was a `Warning: Cannot update a component...while rendering a different component....To locate the bad setState()...`.


## 🧢 Your Project:
